ethers-contract/src/call.rs

```diff
@@ -72,6 +78,34 @@ pub enum ContractError<M: Middleware> {
     ContractNotDeployed,
 }
 
+impl<M: Middleware> ContractError<M> {
+    pub fn decode_revert<Err: EthError>(&self) -> Option<Err> {
+        match self {
+            ContractError::Revert(data) => Err::decode(data).ok(),
+            _ => None,
+        }
+    }
+
+    /// Convert a Middleware Error to a `ContractError`
+    pub fn from_middleware_error(e: M::Error) -> Self {
```

Comment:
can't make `From<M::Error>` because it potentially conflicts with blanket `From<T> for T` in cases where `M::Error` and `ContractError<M>` are the same type

yes this is weird

Reply:
Yeah, this would be fixed with specialization, but alas